From 661202b2fcd7b98f0c40ebbf425da34a6b0097fd Mon Sep 17 00:00:00 2001
From: Lin Fuzhen <fuzhen.lin@freescale.com>
Date: Tue, 21 Aug 2012 16:46:44 +0800
Subject: [PATCH 044/281] ENGR00220895-1 Profile: Add power profile for i.MX6
 SOCs

Change some file permission to support powerprofile.
These files are:
	CPUFREQs
	Thermal

Signed-off-by: Lin Fuzhen <fuzhen.lin@freescale.com>
---
 imx6/etc/ueventd.freescale.rc |  3 +++
 imx6/sabresd/init.rc          | 31 +++++++++++++++++++++++++++++++
 2 files changed, 34 insertions(+)

diff --git a/imx6/etc/ueventd.freescale.rc b/imx6/etc/ueventd.freescale.rc
index ba57391..83a37dc 100644
--- a/imx6/etc/ueventd.freescale.rc
+++ b/imx6/etc/ueventd.freescale.rc
@@ -27,6 +27,9 @@
 /sys/devices/virtual/input/input*   poll        0660  root   input
 /sys/devices/virtual/input/input*   enable      0660  root   input
 /sys/devices/virtual/input/input*   position    0660  root   input
+/sys/devices/virtual/thermal/thermal_zone* trip_point_0_temp 0660 system system
+/sys/devices/virtual/thermal/thermal_zone* trip_point_1_temp 0660 system system
+/sys/devices/virtual/thermal/thermal_zone* trip_point_2_temp 0660 system system
 
 /sys/devices/platform/mxsdhci.2/mmc_host/mmc0/mmc0:000* boot_bus_config 0660 root root
 /sys/devices/platform/mxsdhci.2/mmc_host/mmc0/mmc0:000* boot_config 0660 root root
diff --git a/imx6/sabresd/init.rc b/imx6/sabresd/init.rc
index b2fca75..2977d48 100755
--- a/imx6/sabresd/init.rc
+++ b/imx6/sabresd/init.rc
@@ -52,6 +52,37 @@ on boot
 
     # Set video overlay display mode
     setprop sys.VIDEO_OVERLAY_DISPLAY 1
+# change for FSL specific service
+    chown system system /sys/devices/system/cpu/cpu0
+    chown system system /sys/devices/system/cpu/cpu1
+    chown system system /sys/devices/system/cpu/cpu2
+    chown system system /sys/devices/system/cpu/cpu3
+    chown system system /sys/class/thermal/thermal_zone0
+    chmod 0550 /sys/class/thermal/thermal_zone0
+    chmod 0550 /sys/devices/system/cpu/cpu0
+    chmod 0550 /sys/devices/system/cpu/cpu1
+    chmod 0550 /sys/devices/system/cpu/cpu2
+    chmod 0550 /sys/devices/system/cpu/cpu3
+    chmod 0550 /sys/class/thermal/thermal_zone0
+    chown system system /sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq
+    chmod 0660 /sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq
+    chown system system /sys/devices/system/cpu/cpu0/cpufreq/scaling_min_freq
+    chmod 0660 /sys/devices/system/cpu/cpu0/cpufreq/scaling_min_freq
+    chown system system /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor
+    chown system system /sys/devices/system/cpu/cpu0/cpufreq/cpuinfo_cur_freq
+    chmod 0440 /sys/devices/system/cpu/cpu0/cpufreq/cpuinfo_cur_freq
+    chown system system /sys/devices/system/cpu/online
+    chmod 0440 /sys/devices/system/cpu/online
+    chown system system /sys/devices/system/cpu/offline
+    chmod 0440 /sys/devices/system/cpu/offline
+    chown system system /sys/devices/system/cpu/cpu0/online
+    chmod 0660 /sys/devices/system/cpu/cpu0/online
+    chown system system /sys/devices/system/cpu/cpu1/online
+    chmod 0660 /sys/devices/system/cpu/cpu1/online
+    chown system system /sys/devices/system/cpu/cpu2/online
+    chmod 0660 /sys/devices/system/cpu/cpu2/online
+    chown system system /sys/devices/system/cpu/cpu3/online
+    chmod 0660 /sys/devices/system/cpu/cpu3/online
 
     # Default backlight device
     setprop hw.backlight.dev "pwm-backlight.0"
-- 
1.8.0

